frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -532,6 +549,30 @@ export class TransformResourceData {
     });
   };
 
+  getPipelineRunsForPipeline = (pipeline: Pipeline): PipelineRun[] => {
+    const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+    const pipelineRuns = _.get(this.resources, ['pipelineRuns', 'data']);
+    const pipelineName = pipeline.metadata.name;
+    return pipelineRuns.filter((pr: PipelineRun) => {
+      return (
+        pipelineName ===
+        (_.get(pr, ['metadata', 'labels', PIPELINE_RUN_LABEL], null) ||
+          _.get(pr, ['spec', 'pipelineRef', 'name'], null))
```

Comment:
Interesting, do we need both of these? Looks like `PIPELINE_RUN_LABEL` is always there, but `spec.pipelineRef.name` is really targeted to the goal here.

Based on the docs, it appears the `spec.pipelineRef.name` is not guaranteed. But I don't believe that should be a concern of ours - if they inline a pipeline - it can't be related to this system (which is the collection of pipeline runs of a **specified** pipeline)

```
Since a PipelineRun is an invocation of a Pipeline, you must sepcify what Pipeline to invoke.

You can do this by providing a reference to an existing Pipeline
...
Or you can embed the spec of the Pipeline directly in the PipelineRun
```
Ref: https://github.com/tektoncd/pipeline/blob/master/docs/pipelineruns.md#specifiying-a-pipeline

Reply:
I think we can keep both. As you said `pipelineRef` is really the targeted goal here. We should check for `pipelineRef` first, If that's not available we can turn to the label. WDYT?

Reply:
fixed in https://github.com/openshift/console/pull/3014/commits/657b83830153a7aaf5401872fe3075a6ef553c11

Reply:
As discussed, I think it's okay. I don't think it's needed, but it's not a bad fallback if the spec changes in some way.